### ISSUE TITLE
result&batch_size

### ISSUE THREAD

**Issue by aaaa916** [Franpin/TopoLogic#9]
Hello, while reproducing the code, I noticed in section 4.2 "Implementation Details" of your paper, under the training section, it mentions: “All experiments are trained for 24 epochs on 8 NVIDIA RTX 3090 GPUs with a batch size of 2.” I understand this to mean the batch size is 8*2. However, in the code, there is a mention of base_batch_size=8 when adjusting the learning rate based on the number of GPUs. Currently, I am training on 2 RTX 3090 GPUs, setting the batch size to 1 and 2, and training twice. However, the results still differ from yours. So, I would like to ask how exactly the batch size should be set here, and whether it is possible to achieve the same results with fewer GPUs. The best result I have reproduced so far is shown in the figure. I am using 3 GPUs, with samples_per_gpu=1, so the batch size is 3. All other configuration parameters are the same, and the learning rate has been adjusted according to the number of GPUs. However, I have a question after reviewing the code: Does the topology result serve as input to each decoder layer in the detection part? The reason I ask is that the detection performance does not show improvement compared to previous methods.
![27ec6613c10a192fe10174f596d6a4a](https://github.com/user-attachments/assets/0049f6b8-25a0-469e-bfd6-e5c81e3f8a5c)
Thank you!

**Comment by Franpin** [comment]
Thank you for your interest. First of all, the TopoLogic paper trains with 8 GPUs and each GPU batch size of 2 to save training time. In the code, each GPU batch size is 1. The training results of these two settings are basically the same. Secondly, if your results differ too much from your paper's results, it may be that your mmde3d environment is inconsistent. Finally, the lane topology is sent to the GNN to enhance lane learning.

**Comment by Franpin** [comment]
Thank you for your interest. First of all, the TopoLogic paper trains with 8 GPUs and each GPU batch size of 2 to save training time. In the code, each GPU batch size is 1. The training results of these two settings are basically the same. Secondly, if your results differ too much from results in paper, it may be that your mmde3d environment is inconsistent, you can check you environment. Finally, the lane topology is sent to the GNN to enhance lane learning.

**Comment by aaaa916** [comment]
thanks a  lot

**Comment by aaaa916** [comment]
Hello, according to your README, when I try to reproduce the code, should the result be from the OpenLane V2-v2.1.0 version? The reason I ask is that the result I obtained is: "OpenLane-V2 Score": 0.4066, "DET_l": 0.2539297044277191, "DET_t": 0.45900386571884155, "TOP_ll": 0.19849, "TOP_lt": 0.21896. At first, I thought I was using the evaluation standard from OpenLane V2-2.0.0, so I thought the result was roughly 41.6, but recently when I checked the OpenLaneV2 #77 reference, I found that the OpenLaneV2 package version is 2.1.0, and the evaluation.py is also the same as in version 2.1.0. It seems that the score of 40.6 is the result under v2.1.0, which is much lower than the 44.6 displayed in the original paper.
